### PR TITLE
AppRole with Terraform & Chef

### PR DESCRIPTION
**Pull request by yhyakuna**
This guide walkthrough the audience to reproduce Teddy's webinar demo:
https://github.com/hashicorp/vault-guides/tree/master/identity/vault-chef-approle

Provide a working demo of  _Trusted Entities_ discussion in [AppRole intro guide](https://www.vaultproject.io/guides/identity/authentication.html#advanced-features)

### CODE REVIEW

**Comment by jefferai** [comment]
Hi there,

One higher level architectural question: is it the Chef Node that we want to have access to the secret, or some application or service that the Chef Node is delivering it to? If the latter, the AppRole Secret-ID should be response-wrapped by the Chef Server so that the Node cannot see the actual value, and so that the end application can verify that only it has ever seen the credential.

(There's also a good alternate story here, which Nomad uses, of just using token roles instead of AppRole to reduce overall complexity.)

**Comment by yhyakuna** [comment]
@jefferai  Yes, Teddy talks about SecretID should be wrapped in his webinar, too.  But I guessed he used shortcuts for the sake of his demo, so he added "**NOT SUITABLE FOR PRODUCTION USE**" sign.

File: website/source/guides/identity/approle-trusted-entities.html.md [hunk 1]
```diff
@@ -0,0 +1,779 @@
...
+
+- Chef cookbook (**`/chef`**): A sample cookbook with a recipe that installs NGINX
+and demonstrates Vault Ruby Gem functionality used to interact with Vault APIs.
+- Terraform configurations (**`/terraform-aws`**):
+    - **`/terraform-aws/mgmt-node`**: Configuration to set up a management
+    server running both Vault and Chef Server, for demo purposes.
+    - **`/terraform-aws/chef-node`**: Configuration to set up a Chef node and
+    bootstrap it with the Chef Server, passing in Vault's AppRole RoleID and the
+    appropriate Chef run-list.
+- Vault configuration (**`/vault`**): Data scripts used to configure the
```
Comment (bensojona) [inline website/source/guides/identity/approle-trusted-entities.html.md]: It doesn't look like the `/vault` folder exists

File: website/source/guides/identity/approle-trusted-entities.html.md [hunk 2]
```diff
@@ -0,0 +1,779 @@
...
+
+### Download demo assets
+
+Clone or download the demo assets from the
+[hashicorp/vault-guides](https://github.com/hashicorp/vault-guides/tree/master/identity/vault-chef-approle)
+GitHub repository to perform the steps described in this guide.
+
+The following assets can be found in the repository:
+
+- Chef cookbook (**`/chef`**): A sample cookbook with a recipe that installs NGINX
```
Comment (bensojona) [inline website/source/guides/identity/approle-trusted-entities.html.md]: In the `vault-guides` repo this is `/chef/cookbooks`, may want to update one or the other to be consistent. Unless we're adding more folders under `/chef` we can probably just axe the `cookbooks` sub directory.

File: website/source/guides/identity/approle-trusted-entities.html.md [hunk 3]
```diff
@@ -0,0 +1,779 @@
...
+
+
+### Step 3: AppRole Setup
+
+First, initialize and unseal the Vault server using a shortcut.
+
+~> This is a convenient shortcut for demo. **_DO NOT DO THIS IN PRODUCTION!!!_**
+
+```shell
+# Initialize the Vault server and writ out the unseal keys and root token into files
```
Comment (bensojona) [inline website/source/guides/identity/approle-trusted-entities.html.md]: write

File: website/source/guides/identity/approle-trusted-entities.html.md [hunk 4]
```diff
@@ -0,0 +1,779 @@
...
+[Phase 2 - Provision our Chef Node to Show AppRole Login](#phase2).
+- **Option 2:** Continue onto [Step 3: AppRole Setup](#step-3-approle-setup) to
+set up the demo environment ***manually***.
+
+
+### Step 3: AppRole Setup
+
+First, initialize and unseal the Vault server using a shortcut.
+
+~> This is a convenient shortcut for demo. **_DO NOT DO THIS IN PRODUCTION!!!_**
```
Comment (bensojona) [inline website/source/guides/identity/approle-trusted-entities.html.md]: Maybe we can add a link to the docs for how to unseal using best practices so they can choose to do it the correct way if they want.

File: website/source/guides/identity/approle-trusted-entities.html.md [hunk 5]
```diff
@@ -0,0 +1,779 @@
...
+#### Set up our AppRole policy
+
+This is the policy that will be attached to _secret zero_ which you are
+delivering to our appliaction (**app-1**).
+
+`app-1-secret-read.hcl`:
+
+```bash
+# Grant rad and list capabilities on secret/app-1 path
+path "secret/app-1" {
```
Comment (bensojona) [inline website/source/guides/identity/approle-trusted-entities.html.md]: Can we add the below so they can just copy/paste?

```
cat <<EOF | sudo tee app-1-secret-read.hcl
# Grant rad and list capabilities on secret/app-1 path
path "secret/app-1" {
  capabilities = [ "read", "list"]
}
EOF
```

File: website/source/guides/identity/approle-trusted-entities.html.md [hunk 6]
```diff
@@ -0,0 +1,779 @@
...
+       --data @roleid-token-config.json \
+       $VAULT_ADDR/v1/auth/token/create > roleid-token.json
+```
+
+The token and associated metadata will be written out to the file
+`roleid-token.json`. The `client_token` value is what you'll give to Terraform.
+The file should look similar to the following:
+
+```plaintext
+$ cat roleid-token.json | jq
```
Comment (bensojona) [inline website/source/guides/identity/approle-trusted-entities.html.md]: Do we install jq for them in the script? If not we should make sure it's installed in that userdata script.

File: website/source/guides/identity/approle-trusted-entities.html.md (hunk 1 above)
Comment (yhyakuna) [reply website/source/guides/identity/approle-trusted-entities.html.md]: Thank you for catching my typo.  This should've been `scripts`.

File: website/source/guides/identity/approle-trusted-entities.html.md (hunk 6 above)
Comment (yhyakuna) [reply website/source/guides/identity/approle-trusted-entities.html.md]: I've already tested it all successfully, and `jq` is already on the image.

**Comment by yhyakuna** [comment]
As @bensojona pointed out, I was not following the guide template.    Now, I added matching CLI commands for all API calls just like all other guides.

**Pull request merged**